napari/tests/util.py

```diff
@@ -0,0 +1,89 @@
+import napari
+import numpy as np
+from napari.layers import (
+    Image,
+    Labels,
+    Points,
+    Shapes,
+    Vectors,
+    Surface
+)
+
+layer_test_data = [
+    (Image, np.random.random((10, 15)),  2),
+    (Image, np.random.random((10, 15, 20)), 3),
+    (Image, [np.random.random(s) for s in [(40, 20), (20, 10), (10, 5)]], 2),
+    (Labels, np.random.randint(20, size=(10, 15)), 2),
+    (Points, 20 * np.random.random((10, 2)), 2),
+    (Vectors, 20 * np.random.random((10, 2, 2)), 2),
+    (Shapes, 20 * np.random.random((10, 4, 2)), 2),
+    (Surface, (np.random.random((10, 3)), np.random.randint(10, size=(6, 3)),
+                 np.random.random(10)), 3)
+]
+
+
+def add_layer_by_type(viewer, layer_type, data):
+    """
+    Convenience method that maps a LayerType to it's add_layer method.
+
+    Parameters
+    ----------
+    layer_type : LayerTypes
+        Layer type to add
+    data :
+        The layer data to add
+    """
+    if layer_type is Labels:
+        return viewer.add_labels(data)
+    if layer_type is Points:
+        return viewer.add_points(data)
+    if layer_type is Vectors:
+        return viewer.add_vectors(data)
+    if layer_type is Shapes:
+        return viewer.add_shapes(data)
+    if layer_type is Surface:
+        return viewer.add_surface(data)
+    if layer_type is Image:
+        return viewer.add_image(data)
+
+
+def view_layer_type(layer_type, data):
+    """
+    Convenience method that maps a LayerType to it's view method.
+
+    Parameters
+    ----------
+    layer_type : LayerTypes
+        Layer type to view
+    data :
+        The layer data to view
+    """
+    if layer_type is Image:
+        return napari.view_image(data)
+    if layer_type is Labels:
+        return napari.view_labels(data)
+    if layer_type is Points:
+        return napari.view_points(data)
+    if layer_type is Vectors:
+        return napari.view_vectors(data)
+    if layer_type is Shapes:
+        return napari.view_shapes(data)
+    if layer_type is Surface:
+        return napari.view_surface(data)
+
+
+def check_viewer_functioning(viewer, view, data, ndim):
```

Comment:
can we make `view=None` optional and at end of parameter list and if not provided then we don't make those checks.

Reply:
Yeah, in fact, I might make this whole function keyword-only, since it's unclear in what order these parameters *should* appear. Well, maybe viewer can be positional.

But, btw, I love this function! So much functionality tested for all our layer types now! :rocket: